simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
+        description="""
+        Specifies the maximum number of allowed self-consistent iterations. The simulation `is_converged`
+        if the number of iterations is not larger or equal than this quantity.
+        """,
+    )
+
+    # ? define class `Tolerance` for the different Scf tolerances types?
+    threshold_energy_change = Quantity(
+        type=np.float64,
+        unit="joule",
+        description="""
+        Specifies the threshold for the total energy change between two subsequent self-consistent iterations.
+        The simulation `is_converged` if the total energy change between two SCF cycles is below
+        this threshold.
+        """,
+    )
+
+    threshold_charge_density_change = Quantity(
+        type=np.float64,
+        description="""
+        Specifies the threshold for the average charge density change between two subsequent
+        self-consistent iterations. The simulation `is_converged` if the charge density change
+        between two SCF cycles is below this threshold.
+        """,
+    )
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Set the name of the section
+        self.name = self.m_def.name if self.name is None else self.name
+
+
+class BasisSet(NumericalSettings):
+    """"""
+
+    # TODO work on this base section (@ndaelman-hu)
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Set the name of the section
+        self.name = self.m_def.name if self.name is None else self.name
 
 
 class ModelMethod(ArchiveSection):
-    """ """
+    """
+    Model method input parameters and numerical settings used to simulate the materials properties.
+
+    # ! add more description once section is finished
+    """
 
     normalizer_level = 1
 
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the model method. This is typically used to easy identification of the `ModelMethod` section.
+        Possible values: "DFT", "TB", "BSE", "GW", "DMFT".
+        """,
+        a_eln=ELNAnnotation(component="StringEditQuantity"),
+    )
+
+    type = Quantity(
+        type=str,
+        description="""
+        Identifier used to further specify the type of model Hamiltonian. Example: a TB
+        model can be 'Wannier', 'DFTB', 'xTB' or 'Slater-Koster'. This quantity should be
+        rewritten to a MEnum when inheriting from this class.
+        """,
+        a_eln=ELNAnnotation(component="StringEditQuantity"),
+    )
+
+    external_reference = Quantity(
+        type=str,
+        description="""
+        External reference to the model e.g. DOI, URL.
+        """,
+        a_eln=ELNAnnotation(component="URLEditQuantity"),
+    )
+
+    n_method_references = Quantity(
+        type=np.int32,
+        description="""
+        Number of other `ModelMethod` section references of the current method.
+        """,
+    )
+
+    methods_ref = Quantity(
```

Comment:
1. I think in my mind, `method[-1]` will be always the one used for reference (e.g., in an output), while the others will be part of `method[-1].methods_ref`.
2. Cannot imagine an example where this can occur. But maybe we should simply work on referencing in outputs.py, and then go back to these kind of quantities? 

As I see it, referencing in these sections is purely to make life easier when trying to find the provenance of a simulation. E.g., in the case of adding `TB.orbitals_ref`, this kind of info can be anyways accessed via other methods, but I just decided to add it there for convenience. This does not mean that it is the correct way of doing it.

Reply:
Personally, I don't have a use case for this (yet). At best maybe basis set extrapolations, but I was thinking of using a workflow there. I kind of get @JFRudzinski 's questions, but let me reformulate them in my own words:

@JosePizarro3 Workflows are already tools for provenance. Now, I'm fine adding more, but the distinction on when to use each should be clear (and documented).
Also, how should information be retrieved here? Is the idea that we only show data from the last method? When should a user traverse the method graph? What kind of tools should we offer there?

Reply:
I just decided to take it out for now.

We can decide later along the way when transferring parsers into this new schema.